aptos-move/aptos-replay-benchmark/src/diff.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use aptos_types::{
+    contract_event::ContractEvent,
+    state_store::state_key::StateKey,
+    transaction::{ExecutionStatus, TransactionOutput},
+    write_set::{WriteOp, WriteSet},
+};
+use claims::assert_ok;
+use std::collections::BTreeMap;
+
+/// Different parts of [TransactionOutput] that can be different:
+///   1. gas used,
+///   2. status (must be kept since transactions are replayed),
+///   3. events,
+///   4. writes.
+/// Note that fine-grained comparison allows for some differences to be okay, e.g., using more gas
+/// implies that the fee statement event, the account balance of the fee payer, and the total token
+/// supply are different.
+enum Diff {
+    GasUsed {
+        left: u64,
+        right: u64,
+    },
+    ExecutionStatus {
+        left: ExecutionStatus,
+        right: ExecutionStatus,
+    },
+    Event {
+        left: Option<ContractEvent>,
+        right: Option<ContractEvent>,
+    },
+    WriteSet {
+        left: Option<(StateKey, WriteOp)>,
```

Comment:
hm, statekey needs to be the same, right? since we will have a container of Diffs, so a writeset diff intuitively is for a statekey where Option means one of them didn't actually produce a write for the given statekey.
So I'd have: 
```
key: StateKey,
left: Option<WriteOp>,
right: Option<WriteOp>
```

Reply:
Oh, that's right, way better like that👍